scripts/clean-shrinkwrap.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * This script is in charge of generating the `shrinkwrap` file.
+ *
+ * `npm shrinkwrap` has a bug where it will add optional dependencies
+ * to `npm-shrinkwrap.json`, therefore causing errors if these optional
+ * dependendencies are platform dependent and you then try to build
+ * the project in another platform.
+ *
+ * As a workaround, we keep a list of platform dependent dependencies in
+ * the `platformSpecificDependencies` property of `package.json`,
+ * and manually remove them from `npm-shrinkwrap.json` if they exist.
+ *
+ * See: https://github.com/npm/npm/issues/2679
+ */
+
+/* eslint-disable lodash/prefer-lodash-method */
+/* eslint-disable no-magic-numbers */
+/* eslint-disable no-undefined */
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const _ = require('lodash')
+
+const JSON_INDENT = 2
+const SHRINKWRAP_FILENAME = path.join(__dirname, '..', 'npm-shrinkwrap.json')
+
+const packageInfo = require('../package.json')
+const shrinkwrap = require('../npm-shrinkwrap.json')
+
+/**
+ * @summary Delete a dependency
+ *
+ * @param {*} keys - Keys reference
+ * @param {*} tree - Tree to iterate over
+ * @param {*} onNode - Function to apply
+ * @param {*} parentName - Name of dependent
+ * @param {*} depType - Dependency type (dependencies|requires)
+ *
+ * @example
+ * removeDependency(keys, tree, onNode, parentName)
```

Comment:
Looks like this example is forgetting to pass in the `depType` parameter? :wink: